added some aria attributes

ciaranschutte opened the pull request:
**Description of changes**

added missing aria roles

for some reason storybook still complains about no aria-checked on Radio. but it's wrong.
![aria-wrong](https://user-images.githubusercontent.com/1486054/61671440-aba0a200-acb5-11e9-8d43-e85e945f3fbd.png)


**Type of Change**
- [x] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- [ ] Matches deisgn:
* component sizes, spacing, and styless
* font size, weight, colour
* spelling has been double checked
- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

hlminh2000 commented:
does it maybe wants it as a string?

ciaranschutte commented:
@hlminh2000 no dice :(

hlminh2000 commented:
just noting our side conversation, we found that it wants the `aria-checked` property on the containing div as well :)

joneubank commented on client/uikit/form/Checkbox/index.js:
```diff
@@ -81,7 +81,7 @@ export const StyledCheckbox = styled('div')`
  * Basic checkbox input
  */
-const Checkbox = ({ checked, disabled, onChange }) => {
+const Checkbox = ({ checked, disabled, onChange, ...props }) => {
```
Why not add ariaLabel as a named input and add all these 4 to the propTypes expected by the component?

hlminh2000 replied:
because I am terrible and have used `aria-label` instead of `ariaLabel`, and that's not a valid name. The good thing is that matches with the Dom's prop name

joneubank replied:
True... but if its a declared prop, and required, we could make our tools check for any form elements that forget to include this property instead of relying on accessibility tests to find missing aria-labels...

As a dev, adding `ariaLabel="whatever"` feels natural on a React component

hlminh2000 replied:
it should be in propTypes, can do something like:
```
Checkbox.propTypes = {
  ["aria-label"]: PropTypes.string
}
```
Or use flow

joneubank replied:
Ok sweet. I'm into that.

joneubank commented on client/uikit/form/Radio/index.js:
```diff
@@ -72,17 +72,22 @@ export const StyledRadio = styled('div')`
  * ::after - circle outline
  */
-const Radio = ({ value, disabled, checked, onChange }) => {
+const Radio = ({ value, disabled, checked, onChange, ...props }) => {
```
Same propTypes comment

hlminh2000 commented on client/uikit/form/Checkbox/index.js:
```diff
@@ -105,6 +106,11 @@ const Checkbox = ({ checked, disabled, onChange }) => {
+  ['aria-label']: PropTypes.string.isRequired,
+  checked: PropTypes.bool.isRequired,
+  disabled: PropTypes.bool.isRequired,
+  onChange: PropTypes.func.isRequired,
```
I don't think `disabled` really needs to be required eh? can be false by default, not every checkbox needs to configure this :)

hlminh2000 commented on client/uikit/form/Radio/index.js:
```diff
@@ -97,6 +102,11 @@ const Radio = ({ value, disabled, checked, onChange }) => {
+Radio.propTypes = {
+  ['aria-label']: PropTypes.string.isRequired,
+  checked: PropTypes.bool.isRequired,
+  disabled: PropTypes.bool.isRequired,
```
.

joneubank commented on client/uikit/form/Checkbox/index.js:
```diff
@@ -92,6 +92,7 @@ const Checkbox = ({ checked, disabled, onChange }) => {
         checked={checked}
         disabled={disabled}
         onChange={onChange}
+        aria-label={props['aria-label']}
```
does this put aria-label="undefined" if none is provided?

ciaranschutte replied:
Yes. PropType required. changing to flow type that will be enforced.

joneubank commented on client/uikit/form/Checkbox/index.js:
```diff
@@ -105,6 +106,11 @@ const Checkbox = ({ checked, disabled, onChange }) => {
 };
 
-Checkbox.propTypes = {};
+Checkbox.propTypes = {
```
Following @hlminh2000 's  advice to me... Should we use flow type declaration in the Checkbox definition so that our checker catches errors?

hlminh2000 replied:
Might need to check storybooks config to make sure flow is configured for uikit, but if it is, sure I think that's better

_Sent from my OnePlus5 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

ciaranschutte replied:
gotta love that FastHub

joneubank commented on client/uikit/form/Radio/index.js:
```diff
@@ -97,6 +102,11 @@ const Radio = ({ value, disabled, checked, onChange }) => {
 };
 
-Radio.propTypes = {};
+Radio.propTypes = {
```
Same propTypes -> flow types comment.

ciaranschutte commented:
@joneubank @hlminh2000 
- replaced proptypes with Flow
- changed flow ignore, Flow needs to know about node_modules or imports start throwing errors

storybook works with flow, just add the `// @flow` comment to file

hlminh2000 commented on client/.flowconfig:
```diff
@@ -1,6 +1,4 @@
 [ignore]
-.*/node_modules/*
-
```
hmm shouldn't this be ignored?

ciaranschutte replied:
if its not ignored, flow will throw errors on some imports @hlminh2000
